src/services/Query.ts

```diff
@@ -0,0 +1,33 @@
+import { ApiPromise } from '@polkadot/api'
+import { AccountInfo } from '@polkadot/types/interfaces'
+import BN from 'bn.js'
+import memoryDatabase from '../utils/MemoryDatabase'
+
+type Iquery = {
+  getNonce(api: ApiPromise, address: string): Promise<BN>
+}
+
+const getNonce = async (api: ApiPromise, address: string): Promise<BN> => {
+  const accountInfo: AccountInfo = await api.query.system.account(address)
+  const onchainNonce: BN = accountInfo.nonce.toBn()
+
+  let nonce: BN
+
+  if (!memoryDatabase.hasAddressNonce(address)) {
```

Comment:
nonce management should be a part of tx sending logic, not quering the nonce- since query should just query nonce from the chain